mm2src/coins/lightning.rs

```diff
@@ -365,13 +601,54 @@ impl SwapOps for LightningCoin {
         _min_block_number: u64,
         _uuid: &[u8],
     ) -> Box<dyn Future<Item = (), Error = String> + Send> {
-        unimplemented!()
+        Box::new(futures01::future::ok(()))
     }
 
     fn validate_maker_payment(&self, _input: ValidatePaymentInput) -> ValidatePaymentFut<()> { unimplemented!() }
 
-    fn validate_taker_payment(&self, _input: ValidatePaymentInput) -> ValidatePaymentFut<()> { unimplemented!() }
+    fn validate_taker_payment(&self, input: ValidatePaymentInput) -> ValidatePaymentFut<()> {
+        let payment_hash = try_f!(payment_hash_from_slice(&input.payment_tx)
+            .map_to_mm(|e| ValidatePaymentError::TxDeserializationError(e.to_string())));
+        let payment_hex = hex::encode(payment_hash.0);
 
+        let amt_msat = try_f!(sat_from_big_decimal(&input.amount, self.decimals()));
+
+        let coin = self.clone();
+        let fut = async move {
+            match coin.db.get_payment_from_db(payment_hash).await {
+                Ok(Some(mut payment)) => {
+                    let amount_sent = payment.amt_msat;
+                    // Todo: Add more validations if needed, locktime is probably the most important
+                    if amount_sent != Some(amt_msat as i64) {
+                        // Free the htlc to allow for this inbound liquidity to be used for other inbound payments
+                        coin.channel_manager.fail_htlc_backwards(&payment_hash);
```

Comment:
I assume there might be other places to call `fail_htlc_backwards` in case of other swap steps failures.

Reply:
Already added to checklist https://github.com/KomodoPlatform/atomicDEX-API/pull/1497#discussion_r1003174281 :)